### DISCUSSION TITLE
Bug avec Synology container manager

### DISCUSSION THREAD

**Discussion by 327-LIVE** [LsHallo/ets2-dedicated-convoy-server#18] [General]
Bonjours, depuis la version 0.3, il est impossible de créé un container. en effet l'importation se fait correctement mais bloque au premier lancement et même en le réinitialisent. 

voici une parti des logs en version 0.2.2 qui fonctionne correctement
Linux version 4.4.302+
Docker compose 2.5
00:00:00.000 : [sys] running on x86_64 / Linux
00:00:00.000 : [sys] Command line:  /app/bin/linux_x64/eurotrucks2_server
00:00:00.000 : [cpu] CPU0: AuthenticAMD [AMD Ryzen Embedded V1500B                      ] with 8 cores (8 threads) at ~2199MHz.
00:00:00.000 : [sys] using 6 worker thread(s)
00:00:00.000 : [sys] and 3 worker threads for IO.
00:00:00.000 : [sys] plus 3 real-time worker threads.
00:00:00.000 : [sys] using CGT CLOCK_MONOTONIC_RAW timer with 1nsec resolution.
00:00:00.000 : [mem] physical total: 32071M
00:00:00.000 : [mem] physical avail: 398M
00:00:00.000 : [mem] virtual total: 17592186044415M
00:00:00.000 : [mem] virtual avail: 17592186044415M
00:00:00.000 : [mem] Trying to allocate memory pool (10485760K)
00:00:00.000 : [mem] Allocated 5 temporary loading buffer(s)
00:00:00.011 : [hashfs] effect.scs: Created and validated, 16126 entries (433F9318568BC2FC0E0701EE600665B635AE1B2965073B8155B53BB965A35F52)
00:00:00.011 : [fs] device effect.scs mounted.
00:00:00.012 : [hashfs] base.scs: Created and validated, 1534 entries (D391E184BF776C25AB64D1A56E2B4D337831E6B8063772B2C38F3AE05F7E84EC)
00:00:00.012 : [fs] device base.scs mounted.
00:00:00.013 : [hashfs] core.scs: Created and validated, 9 entries (BD969A94A40234F8BEC3628A2B65CC3B1F202657D37364524E9D321A942E065A)
00:00:00.013 : [fs] device core.scs mounted.
00:00:00.073 : [hashfs] def.scs: Created and validated, 43123 entries (52B7A3BE7699A068453EE6DEDC4E1ABFFACB25849918BEE64B680A5837506A6F)
00:00:00.073 : [fs] device def.scs mounted.
00:00:00.136 : [hashfs] locale.scs: Created and validated, 262 entries (E632D0A46B99F38EF4C390D1BAA334B168FB70B01382D563BFD6B974061EA0EE)
00:00:00.136 : [fs] device locale.scs mounted.
00:00:00.136 : [hashfs] base_cfg.scs: Created and validated, 1 entries (1F35E79D4CDD82701781E7D0097C86F21D2B453C8E7D84495D5EC0852A19E287)
00:00:00.136 : [fs] device base_cfg.scs mounted.
00:00:00.136 : [ufs] Home directory: '/home/steam/.local/share/Euro Truck Simulator 2'.
00:00:00.136 : [fs] device /home/steam/.local/share/Euro Truck Simulator 2 mounted.
00:00:00.137 : exec /home/config_ds.cfg
00:00:00.137 : [sys] Executing /home/config_ds.cfg ...

Le serveur démarre

**Comment by LsHallo** [comment]
Salut,

please report your issues in english. I can't assist you in french.

I just tried on my server which isn't a synology and it works as expected.
This is the log output from the second start: https://gist.github.com/LsHallo/9aca431d77b809163c6e7379fe4342ba
The line `[sys] Executing /home/config_ds.cfg` also appears for me and the first start it fails with `no such file or directory`.
On the second start it works normally.

Do you have the ets server directory mounted to the file system in the nas like this?
```
-v "/opt/ets-server/:/app"
-v "/opt/ets-server/config:/home/steam/.local/share/Euro Truck Simulator 2"
```

If so please remove all volume mounts and try again. And also share how you run the container (all config files).

**Reply by LsHallo** [reply]
I did change the base image for v0.3.0 if I remember correctly. That might have something to do with your problems.

**Reply by 327-LIVE** [reply]
je vais regarder cela :)

**Reply by 327-LIVE** [reply]
Effectivement le problème se produit depuis la version 0.3.0

**Reply by 327-LIVE** [reply]
voici le système de configuration sur un nas Synology
![config](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/164751161/396aea0a-b678-4bcf-ba4a-039b3324e755)

**Reply by LsHallo** [reply]
Okay then the problem is very likely caused by file permissions since the user id changed between image versions v0.2.x and v0.3.x.
You can try and chown the data directory on your host machine to the new user id: `chown -R 1000:1000 path/to/datadir/of/ets2/server`.
Otherwise it would be wise to delete the server data and start fresh. If you have mods you can copy them to the new data dir after. Make sure to adjust the owner to id 1000 when copying.

**Reply by 327-LIVE** [reply]
Comme vous pouvez le voir, je suis très limiter avec l'interface graphique et ne suis pas dans mon élément avec les commandes telnet. De plus je ne possède surement pas les connaissances pour pouvoir effectuer cela. Je précise que le server est vierge et sans mod.

Vous me dite que l'utilisateur a changer depuis la version 0.3.0 ?
Y a t'il une contrainte technique qui empêcherais de remettre l'utilisateur comme sur la version 0.2.2 ?

**Reply by LsHallo** [reply]
If you don't have any mods installed you can just remove the volume mount completely and use the server without any volume mount. It will work fine that way. No mount is needed for operation without mods. No game state is saved on the server. You can use the server without data loss without using a volume.
Just disregard the volume mount completely. I've also changed the default example to not include the volume mount any more since it is not needed.
So at the top of you config remove all "mappez volumes" and you can use the server without any problem.

**Reply by 327-LIVE** [reply]
j'ai bien compris cette méthode. j'ai cependant besoin d'une sauvegarde des log et pouvoir charger mes paramètres du fichier server_config.sii, comme mon token steam mon nom de serveur et réglages.

**Reply by LsHallo** [reply]
You can use the environment variables to configure all settings of server_settings.sii including logon token. No need to edit the configuration by hand. See README.md for the names of all variables and how to use them.

Logs i can't help you with at this time. But all the server log should be output in the terminal view of the container. It just wont be saved on updates of the container.

**Reply by 327-LIVE** [reply]
Bonjour a vous, je vous remercie pour les réponses que vous m'avez apporté.

Je vais sagement rester en version 0.2.2 qui conviens a mes habitudes et usages "fichier log sauvegarder automatiquement et configuration dans les fichiers du jeu".

Encore merci pour se que vous avez fait avec se projet. 

j'aurais sans doute 2-3 idées a vous communiquer pour les prochaines version, en espèrent pouvoir en profiter sur mon nas comme la 0.2.2 ;)

**Reply by 327-LIVE** [reply]
Bonjours a vous.

Je me permet de revenir ici pour poster une solution si d'autres personne veulent comme moi, héberger votre containeur Docker sur un NAS Synology tout en respectent la configurationnel affin d'avoir l'accès au dossier et fichier.

Dans File Station, faire un clic droits sur le dossier Docker et propriétés, puis permissions,  cliquer sur le bouton crée, puis dans la case utilisateur ou groupe ajouter system et mettre les paramètre que sur la capture ci-jointe. 

Valider les changement et réinitialiser les containeurs. ;)

![Docker proprietes](https://github.com/LsHallo/ets2-dedicated-convoy-server/assets/164751161/7c28a6dd-7699-458c-a12e-263f98c0b7a6)

**Comment by 327-LIVE** [comment]
Désoler mais je ne parle pas englais moi non plus ;)

voici les écrits de la console car le fichier log ne peux ètre créé avec la version 0.3.2 au second démarrage. 
Les dossier sont montés correctement dans les paramètres. 

[INFO]: Starting server...                                                             
*** ERROR *** : [fs] Failed to open file '/home/steam/.local/share/Euro Truck Simulator
 2/server.log.txt' in the create_truncate mode.                                        
[sys] running on x86_64 / Linux                                                        
[sys] Command line:  /app/bin/linux_x64/eurotrucks2_server                             
[cpu] CPU0: AuthenticAMD [AMD Ryzen Embedded V1500B                      ] with 8 cores
 (8 threads) at ~2199MHz.                                                              
[sys] using 6 worker thread(s)                                                         
[sys] and 3 worker threads for IO.                                                     
[sys] plus 3 real-time worker threads.                                                 
[sys] using CGT CLOCK_MONOTONIC_RAW timer with 1nsec resolution.                       
[mem] physical total: 32071M                                                           
[mem] physical avail: 246M                                                             
[mem] virtual total: 17592186044415M                                                   
[mem] virtual avail: 17592186044415M                                                   
[mem] Trying to allocate memory pool (10485760K)                                       
[mem] Allocated 5 temporary loading buffer(s)                                          
[hashfs] effect.scs: Created and validated, 16126 entries (433F9318568BC2FC0E0701EE6006
65B635AE1B2965073B8155B53BB965A35F52)                                                  
[fs] device effect.scs mounted.                                                        
[hashfs] base.scs: Created and validated, 1534 entries (D391E184BF776C25AB64D1A56E2B4D3
37831E6B8063772B2C38F3AE05F7E84EC)                                                     
[fs] device base.scs mounted.                                                          
[hashfs] core.scs: Created and validated, 9 entries (BD969A94A40234F8BEC3628A2B65CC3B1F
202657D37364524E9D321A942E065A)                                                        
[fs] device core.scs mounted.          
[hashfs] def.scs: Created and validated, 43123 entries (52B7A3BE7699A068453EE6DEDC4E1AB
FFACB25849918BEE64B680A5837506A6F)                                                     
[fs] device def.scs mounted.                                                           
[hashfs] locale.scs: Created and validated, 262 entries (E632D0A46B99F38EF4C390D1BAA334
B168FB70B01382D563BFD6B974061EA0EE)                                                    
[fs] device locale.scs mounted.                                                        
[hashfs] base_cfg.scs: Created and validated, 1 entries (1F35E79D4CDD82701781E7D0097C86
F21D2B453C8E7D84495D5EC0852A19E287)                                                    
[fs] device base_cfg.scs mounted.                                                      
[ufs] Home directory: '/home/steam/.local/share/Euro Truck Simulator 2'.               
[fs] device /home/steam/.local/share/Euro Truck Simulator 2 mounted.                   
exec /home/config_ds.cfg                                                               
[sys] Executing /home/config_ds.cfg ...   

*** ERROR *** : [fs] Failed to open file '/home/config_ds.cfg' in the create_truncate m
ode.                                                                                   
*** WARNING *** : [cvar] Can not create file '</home/steam/.local/share/Euro Truck Simu
lator 2 | /home>/config_ds.cfg'.                                                       
*** ERROR *** : [fs] Failed to open file '/home/config_ds.cfg' in the create_truncate m
ode.                                                                                   
*** WARNING *** : [cvar] Can not create file '</home/steam/.local/share/Euro Truck Simu
lator 2 | /home>/config_ds.cfg'.                                                       
[lang] Language autodetection: 0                                                       
Selected language: en_gb                                                               
Euro Truck Simulator 2 init ver.1.49.2.0s (rev. 7c4a5dd550fa)                          
*** ERROR *** : [fs] Failed to open file '/home/config_ds.cfg' in the create_truncate m
ode.                                                                                   
*** WARNING *** : [cvar] Can not create file '</home/steam/.local/share/Euro Truck Simu
lator 2 | /home>/config_ds.cfg'. 

server_cfg                                                                             
*** ERROR *** : [fs] Failed to open file '/home/server_config.sii' in the create_trunca
te mode.                                                                               
*** ERROR *** : save_unit_tree() - Failed to create file (/home/server_config.sii)

**Reply by LsHallo** [reply]
Veuillez essayer de démarrer sans monter de dossier. Ne montez aucun dossier et réessayez.

**Reply by 327-LIVE** [reply]
sans monter de dossier, le serveur démarre avec ses réglages de base

[INFO]: Starting server...                                                             
[sys] running on x86_64 / Linux                                                        
[sys] Command line:  /app/bin/linux_x64/eurotrucks2_server                             
[cpu] CPU0: AuthenticAMD [AMD Ryzen Embedded V1500B                      ] with 8 cores
 (8 threads) at ~2199MHz.  
[sys] using 6 worker thread(s)                                                         
[sys] and 3 worker threads for IO.                                                     
[sys] plus 3 real-time worker threads.                                                 
[sys] using CGT CLOCK_MONOTONIC_RAW timer with 1nsec resolution.                       
[mem] physical total: 32071M                                                           
[mem] physical avail: 275M                                                             
[mem] virtual total: 17592186044415M                                                   
[mem] virtual avail: 17592186044415M                                                   
[mem] Trying to allocate memory pool (10485760K)                                       
[mem] Allocated 5 temporary loading buffer(s)                                          
[hashfs] effect.scs: Created and validated, 16126 entries (433F9318568BC2FC0E0701EE6006
65B635AE1B2965073B8155B53BB965A35F52)                                                  
[fs] device effect.scs mounted.                                                        
[hashfs] base.scs: Created and validated, 1534 entries (D391E184BF776C25AB64D1A56E2B4D3
37831E6B8063772B2C38F3AE05F7E84EC)                                                     
[fs] device base.scs mounted.                                                          
[hashfs] core.scs: Created and validated, 9 entries (BD969A94A40234F8BEC3628A2B65CC3B1F
202657D37364524E9D321A942E065A) 
[fs] device core.scs mounted.                                                          
[hashfs] def.scs: Created and validated, 43123 entries (52B7A3BE7699A068453EE6DEDC4E1AB
FFACB25849918BEE64B680A5837506A6F)                                                     
[fs] device def.scs mounted.                                                           
[hashfs] locale.scs: Created and validated, 262 entries (E632D0A46B99F38EF4C390D1BAA334
B168FB70B01382D563BFD6B974061EA0EE)                                                    
[fs] device locale.scs mounted.                                                        
[hashfs] base_cfg.scs: Created and validated, 1 entries (1F35E79D4CDD82701781E7D0097C86
F21D2B453C8E7D84495D5EC0852A19E287)                                                    
[fs] device base_cfg.scs mounted.                                                      
[ufs] Home directory: '/home/steam/.local/share/Euro Truck Simulator 2'.               
[fs] device /home/steam/.local/share/Euro Truck Simulator 2 mounted.                   
exec /home/config_ds.cfg                                                               
[sys] Failed to execute (/home/config_ds.cfg), the file does not exist.                
storing cvars                                                                          
storing cvars                                                                          
[lang] Language autodetection: 0                                                       
Selected language: en_gb                                                               
Euro Truck Simulator 2 init ver.1.49.2.0s (rev. 7c4a5dd550fa)                          
storing cvars
